### PR TITLE
Jakarta Persistence 3.2: 5th batch

### PR DESCRIPTION
**Pull request by Tomas-Kraus**
- jakartaee/persistence#431 - add SchemaManager
- jakartaee/persistence#490 - add getReference(detachedEntity) to EntityManager
- jakartaee/persistence#451 - add PersistenceUnitUtil.getVersion()
- jakartaee/persistence#431 - FindOption/LockOption/RefreshOption support in EntityManager

### CODE REVIEW

**Comment by rfelcman** [comment]
Sorry but command/test `mvn -V verify -pl :org.eclipse.persistence.jpa.testapps.persistence32 -P oracle` doesn't pass against Oracle23C. See attached reports, mainly `SchemaManagerValidateOnMissingColumnTest>JUnitTestCase.runBare:702->testValidateOn`
[test_persistence32.tar.gz](https://github.com/eclipse-ee4j/eclipselink/files/13646113/test_persistence32.tar.gz)

**Comment by Tomas-Kraus** [comment]
Fixed tests with Oracle DB:
```
[INFO] Tests run: 72, Failures: 0, Errors: 0, Skipped: 0
```

File: foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/config/PersistenceUnitProperties.java [hunk 1]
```diff
@@ -3044,6 +3054,39 @@ public class PersistenceUnitProperties {
      */
     public static final String DDL_GENERATION_INDEX_FOREIGN_KEYS = "eclipselink.ddl-generation.index-foreign-keys";
 
+    /**
+     * The "<code>eclipselink.schema-validation.mode</code>" property specifies database schema validation mode.
+     * By default, the "<code>simple</code>" validation mode is selected.
+     * <b>Allowed Values:</b>
+     * <ul>
+     * <li>"<code>simple</code>" (DEFAULT)
+     * <li>"<code>full</code>" (experimental feature, may not work properly)
```
Comment (lukasj) [inline foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/config/PersistenceUnitProperties.java:3063]: just say it is experimental feature

File: foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/descriptors/VersionLockingPolicy.java [hunk 2]
```diff
@@ -547,7 +547,7 @@ public boolean isStoredInObject() {
      * INTERNAL:
      * Retrieves the lock value from the object.
      */
-    protected Object lockValueFromObject(Object domainObject) {
+    public Object lockValueFromObject(Object domainObject) {
```
Comment (lukasj) [inline foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/descriptors/VersionLockingPolicy.java:550]: add new `getVersionValue` method

File: foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/databaseaccess/DatabasePlatform.java [hunk 3]
```diff
@@ -3802,7 +3802,34 @@ public void initializeConnectionData(Connection connection) throws SQLException
      */
      public void writeAddColumnClause(Writer writer, AbstractSession session, TableDefinition table, FieldDefinition field) throws IOException {
         writer.write("ADD ");
-        field.appendDBString(writer, session, table);
+        field.appendDBCreateString(writer, session, table);
+    }
+
+    /**
+     * INTERNAL:
+     * May need to override this method if the platform supports ALTER TABLE DROP COLUMN &lt;column&gt;
```
Comment (lukasj) [inline foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/databaseaccess/DatabasePlatform.java:3810]: use `{@code ...}` or similar

File: foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/tools/schemaframework/SchemaManager.java [hunk 4]
```diff
@@ -1028,8 +1033,8 @@ public void createDefaultTables(boolean generateFKConstraints) {
         try {
             TableCreator tableCreator = getDefaultTableCreator(generateFKConstraints);
             tableCreator.createTables(this.session, this);
-        } catch (DatabaseException ex) {
-            // Ignore error
+        } catch (DatabaseException exception) {
+            session.log(SessionLog.WARNING, SessionLog.DDL, "schema_default_create_tables_failed", exception.getLocalizedMessage());
```
Comment (lukasj) [inline foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/tools/schemaframework/SchemaManager.java:1037]: use finest

File: foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/databaseaccess/DatabasePlatform.java (hunk 3 above)
Comment (Tomas-Kraus) [reply foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/databaseaccess/DatabasePlatform.java]: done

File: foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/tools/schemaframework/SchemaManager.java (hunk 4 above)
Comment (Tomas-Kraus) [reply foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/tools/schemaframework/SchemaManager.java]: done

File: foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/descriptors/VersionLockingPolicy.java (hunk 2 above)
Comment (Tomas-Kraus) [reply foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/descriptors/VersionLockingPolicy.java]: done

File: foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/config/PersistenceUnitProperties.java (hunk 1 above)
Comment (Tomas-Kraus) [reply foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/config/PersistenceUnitProperties.java]: done

**Review by lukasj — APPROVED** [review]

**Pull request merged**